### DISCUSSION TITLE
Installing: `bashly completions --install` in a non-root environment

### DISCUSSION THREAD

**Discussion by b9Joker108** [bashly-framework/bashly#573] [Q&A]
Is it somehow possible to install: `bashly completions --install` in a non-rooted device, user or environment? Moreover, I am confused as to why completions necessitate sudo privileges and if someone could explain this to me, I would be most appreciative.

**Comment by DannyBen** [comment]
Well, bashly delegates its completion actions to the Completely gem, here is the relevant line from the bashly completions command:

https://github.com/DannyBen/bashly/blob/1e567aa547558b8bd1c62709a287cd7d5c8c003c/lib/bashly/commands/completions.rb#L26

Then, the completely installer looks for possible directories to install to - some of which are outside of the user control:

https://github.com/DannyBen/completely/blob/c413d319e9bf9f9b82e7caa8d3909825ed4cb40d/lib/completely/installer.rb#L10-L15

So:

1. You can simply install (copy) the completions script manually to any of your completions directories
2. If there is a better behavior to be implemented here, I am open to it

Perhaps adding a `--user` flag, to try and install only in the two user directories, and therefore without sudo?
